src/S3/Exception/S3MultipartUploadException.php

```diff
@@ -0,0 +1,54 @@
+<?php
+namespace Aws\S3\Exception;
+
+use Aws\CommandInterface;
+use Aws\Exception\AwsException;
+use Aws\Multipart\UploadState;
+
+class S3MultipartUploadException extends \Aws\Exception\MultipartUploadException
+{
+    /** @var string File path of transfer object */
+    private $filePath;
+
+    /**
+     * @param UploadState      $state Upload state at time of the exception.
+     * @param \Exception|array $prev  Exception being thrown. Could be an array of
+     *                                AwsExceptions being thrown when uploading parts
+     *                                for one object, or an instance of AwsException
+     *                                for a specific Multipart error being thrown in
+     *                                the MultipartUpload process.
+     */
+    public function __construct(UploadState $state, $prev = null) {
+        if (is_array($prev)) {
+            foreach ($prev as $part => $error) {
+                $this->collectPathInfo($error->getCommand());
+            }
+        } elseif ($prev instanceof AwsException) {
+            $this->collectPathInfo($prev->getCommand());
+        }
+        parent::__construct($state, $prev);
+    }
+
+    /**
+     * Get file path of S3 transfer
+     *
+     * @return string|null Returns null when 'Bucket' or 'Key' information
+     *                     are unavailable.
+     */
+    public function getFilePath()
```

Comment:
Is there a reason to not just have a getBucket and getKey method instead of returning these as a single concatenated string?